content/shared/influxdb3-processing-engine.md

```diff
@@ -0,0 +1,86 @@
+The Processing engine is an embedded Python virtual machine that runs inside an {{% product-name %}} database server. It executes Python code in response to triggers and database events without requiring external application servers or middleware.
+
+## How it works
+
+### Architecture
+
+The Processing engine runs Python code directly within a {{% product-name %}} server process. This design provides high performance and direct access to database resources.
+
+- **Embedded execution**: Code runs in the same process space as the database server
+- **Direct data access**: Zero-copy access to data
+- **Event-driven**: Responds to database writes, scheduled events, and HTTP requests
+- **Cache integration**: Access to system caches including Last values  and Distinct 
+values 
+
+> [!Note]
+> The Processing engine runs all plugins in the same Python process. Changes made by one plugin can affect other plugins, 
+> as the system does not fully isolate changes from each other. 
```

Comment:
```suggestion
> The Processing engine runs all plugins in the same Python process. Changes made by one plugin can affect other plugins.
```

I just noticed that the end of this sentence isn't correct, and probably isn't necessary unless we have more specifics.